### PR TITLE
Fix invalidate children build when deleting parent node

### PR DESCRIPTION
**Pull request by YenguiSeddik**

### CODE REVIEW

File: src/test/java/org/gridsuite/study/server/NetworkModificationTreeTest.java [hunk 1]
```diff
@@ -67,16 +71,34 @@
...
+import static org.junit.Assert.assertTrue;
 import static org.mockito.ArgumentMatchers.isA;
 import static org.mockito.BDDMockito.given;
 import static org.mockito.Mockito.doNothing;
-import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
```
Comment (AbdelHedhili) [inline src/test/java/org/gridsuite/study/server/NetworkModificationTreeTest.java:102]: do not expand all * import please

**Review by AbdelHedhili — CHANGES_REQUESTED** [review]

File: src/test/java/org/gridsuite/study/server/NetworkModificationTreeTest.java [hunk 2]
```diff
@@ -410,7 +443,8 @@ public void testNodeManipulation() throws Exception {
         children = child.getChildren();
         assertChildrenEquals(Set.of(node1, node2), children);
 
-        deleteNode(root.getStudyId(), child, false, Set.of(child));
+        child.getChildren().forEach(node -> System.out.println(node.getId()));
```
Comment (laurgarn) [inline src/test/java/org/gridsuite/study/server/NetworkModificationTreeTest.java:446]: Printf to remove, I guess

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 3]
```diff
@@ -1343,6 +1344,8 @@ public void deleteNode(UUID studyUuid, UUID nodeId, boolean deleteChildren) {
         startTime.set(System.nanoTime());
         DeleteNodeInfos deleteNodeInfos = new DeleteNodeInfos();
         deleteNodeInfos.setNetworkUuid(networkStoreService.doGetNetworkUuid(studyUuid));
+        boolean invalidateChildrenBuild = !EMPTY_ARRAY.equals(networkModificationTreeService.getNetworkModifications(studyUuid, nodeId));
```
Comment (AbdelHedhili) [inline src/main/java/org/gridsuite/study/server/service/StudyService.java:1347]: you didn't add it, but it's weird getNetworkModifications takes a studyUuid parameter that it never uses

File: src/test/java/org/gridsuite/study/server/NetworkModificationTreeTest.java (hunk 1 above)
Comment (YenguiSeddik) [reply src/test/java/org/gridsuite/study/server/NetworkModificationTreeTest.java]: ok. done

File: src/test/java/org/gridsuite/study/server/NetworkModificationTreeTest.java (hunk 2 above)
Comment (YenguiSeddik) [reply src/test/java/org/gridsuite/study/server/NetworkModificationTreeTest.java]: Done

**Review by AbdelHedhili — APPROVED** [review]

**Review by laurgarn — APPROVED** [review]

**Pull request merged**